File: tiny-fests-api/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;

class Event extends Model
{
    use HasFactory;

    protected $fillable = [
        'title', 'location', 'start_time', 'description', 'image', 'max_spots',
    ];

    protected $casts = [
        'start_time' => 'datetime',
    ];
}

File: tiny_fests_app/lib/models/event.dart
class Event {
  final int? id;
  final String title;
  final String location;
  final DateTime startTime;
  final String description;
  final String? image;
  final int maxSpots;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  Event({
    this.id,
    required this.title,
    required this.location,
    required this.startTime,
    required this.description,
    this.image,
    required this.maxSpots,
    this.createdAt,
    this.updatedAt,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      title: json['title'],
      location: json['location'],
      startTime: DateTime.parse(json['start_time']),
      description: json['description'],
      image: json['image'],
      maxSpots: json['max_spots'],
      createdAt: json['created_at'] != null ? DateTime.parse(json['created_at']) : null,
      updatedAt: json['updated_at'] != null ? DateTime.parse(json['updated_at']) : null,
    );
  }
}